polkadot/node/network/statement-distribution/src/v2/requests.rs

```diff
@@ -528,6 +529,7 @@ impl UnhandledResponse {
 	///   * If the response is partially valid, misbehavior by the responding peer.
 	///   * If there are other peers which have advertised the same candidate for different
 	///     relay-parents or para-ids, misbehavior reports for those peers will also be generated.
+	///   * Statements from disabled validators are filtered out and reported as misbehavior.
```

Comment:
Note that this leads to a race condition, where the requester has witnessed a validator as disabled but the responder hasn't. Because the requester should add disabled validators to the request's exclusion mask, if the requester does not do that it's because the disabled validator was just disabled and the remote peer shouldn't be punished for learning about that marginally later. If the disabled validators are in the request's exclusion mask, the extra statements can be treated just as any other unexpected statements and ignored.

In the case that the responder has learned about the disabled validator before the requester, this is handled by the suggestion in https://github.com/paritytech/polkadot-sdk/pull/1841/files#r1389797262